### PR TITLE
update to nsb v7

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@Particular/azure-maintainers so a bunch of acceptance tests failed. but it seems many acc tests were previously excluded. https://github.com/Particular/NServiceBus.Persistence.ServiceFabric/blob/develop/src/NServiceBus.Persistence.ServiceFabric.AcceptanceTests/AcceptanceTests.csproj#L29

how did u work out what tests were invalid and should be excluded? who can help me go through that process again?

**Comment by danielmarbach** [comment]
@SimonCropp The idea was that we only let the outbox and the saga acceptance tests run because only those are relevant for this persister. I can help you finish this PR.

**Comment by SimonCropp** [comment]
@danielmarbach 

> I can help you finish this PR.

that would be awesome.

**Comment by danielmarbach** [comment]
@SimonCropp on my machine I get the following compile error:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS7069	Reference to type 'ExposeSettings' claims it is defined in 'NServiceBus.Core', but it could not be found	AcceptanceTests	C:\p\NServiceBus.Persistence.ServiceFabric\src\NServiceBus.Persistence.ServiceFabric.AcceptanceTests\ConfigureEndpointAzureServiceBusTransport.cs	28	Active

```

**Comment by danielmarbach** [comment]
Ok apparently I had a custom build called 8.0.0-strategies in my local nuget feed which took precedence over the latest alpha. Now that this is fixed I'm debugging why the tests are failing.

**Comment by danielmarbach** [comment]
@SimonCropp got it down to a single test failing. The approval test does not seem to work yet. Due to locking issues we are moving the sources to C:\temp on the build agent. For some reasons when the tests are executed the following happens:

* The received file lands under `C:\temp\APIApprovals.ApproveServiceFabricPersistence.received.txt`
* The approved file is checked in under `\Tests\APIApprovals.ApproveServiceFabricPersistence.approved.txt` and therefore will reside under `C:\temp\Tests\APIApprovals.ApproveServiceFabricPersistence.approved.txt`

Any ideas how to properly resolve this? Because I can't get it to run locally either.

**Comment by danielmarbach** [comment]
Btw @SimonCropp @Particular/azure-maintainers the tests were failing because we are using ASB and ASB requires the static logger configuration since the AsyncLocal will not be set inside the message pump. 

@Particular/nservicebus-maintainers I'm starting to wonder whether using [AsyncLocal ](https://github.com/Particular/NServiceBus/blob/a91582680cd936d14a583efc583bd7133deee242/src/NServiceBus.AcceptanceTesting/ScenarioContext.cs#L48) to store the scenario context was ever a good idea since any pump action that is happing in a different context that the scenario runner will never see the scenario context being set. So essentially any transport could run into the problem of the scenario context being null and I'm wondering whether having a static context is the lesser of the two evil.

**Comment by SimonCropp** [comment]
@danielmarbach build is green  :)

File: src/Tests/SetUpFixture.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using System.Runtime.CompilerServices;
+using NUnit.Framework;
+
+[SetUpFixture]
+public class SetUpFixture
+{
+    [OneTimeSetUp]
+    public void SetUp()
+    {
+        FixCurrentDirectory();
```
Comment (danielmarbach) [inline src/Tests/SetUpFixture.cs]: Ah nice!

File: src/Tests/SetUpFixture.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    }
+
+    void FixCurrentDirectory([CallerFilePath] string callerFilePath="")
```
Comment (andreasohlund) [inline src/Tests/SetUpFixture.cs]: Ninja! 👍

**Pull request merged**